doc/api/url.md

```diff
@@ -250,7 +250,407 @@ properties of URL objects:
 For example, the ASCII space character (`' '`) is encoded as `%20`. The ASCII
 forward slash (`/`) character is encoded as `%3C`.
 
+## The WHATWG URL API
+
+> Stability: 1 - Experimental
+
+The `url` module provides an experimental implementation of the
+[WHATWG URL Standard][] as an alternative to the existing `url.parse()` API.
+
+```js
+const URL = require('url').URL;
+const myURL = new URL('https://example.org/foo');
+
+console.log(myURL.href);     // "https://example.org/foo"
+console.log(myURL.protocol); // "https:"
+console.log(myURL.hostname); // "/example.org"
+console.log(myURL.pathname); // "/foo"
+```
+
+### Class: URL
+#### Constructor: new URL(input[, base])
+
+* `input` {String} The input URL to parse
+* `base` {String | URL} The base URL to resolve against if the `input` is not
+  absolute.
+
+Creates a new `URL` object by parsing the `input` relative to the `base`. If
+`base` is passed as a string, it will be parsed equivalent to `new URL(base)`.
+
+```js
+const myURL = new URL('/foo', 'https://example.org/');
+  // https://example.org/foo
```

Comment:
Just a style I generally prefer (and have used in other docs that show the expected output